capstone/src/main/webapp/youtubeScript.js

```diff
@@ -86,28 +85,47 @@ async function inputCommentsToPerspective(commentsList) {
   if (!requestedAttributes) {
       return;
   }
-  const attributeTotals = new Map();
+  const attributeScoresPromises = [];
   for (const comments in commentsList) {
-    const attributeScores = [];
     for (const item in commentsList[comments].items) {
-      const perspectiveScore = await callPerspective(commentsList[comments].items[item].snippet.topLevelComment.snippet.textOriginal, langElement.value, requestedAttributes);
-      attributeScores.push(perspectiveScore);
+      let commentText = commentsList[comments].items[item].snippet.topLevelComment.snippet.textOriginal;
+      const perspectiveScore = await callPerspective(commentText, langElement.value, requestedAttributes);
+      attributeScoresPromises.push(perspectiveScore);
     }
-    for (var i = 0; i < requestedAttributes.length; i++) {
-      for (var j = 0; j < attributeScores.length; j++) {
-        if (attributeTotals.has(requestedAttributes[i])) {
-          attributeTotals.set(requestedAttributes[i], attributeTotals.get(requestedAttributes[i]) + attributeScores[j].attributeScores[requestedAttributes[i]].summaryScore.value);
-        } else {
-          attributeTotals.set(requestedAttributes[i], attributeScores[j].attributeScores[requestedAttributes[i]].summaryScore.value);
-        }
+  }
+  let totalNumberOfComments = commentsList[0].items.length * commentsList.length;
```

Comment:
Just curious, are all the `items` in each entry of `commentsList` expected to be the same length?

Reply:
Yes, the length of the items in each entry of commentsList is dependent on the servlet response. The lengths are expected to be the same unless there is an API error which would have been caught before this point